webapp/channels/src/packages/mattermost-redux/src/actions/users.ts

```diff
@@ -149,15 +150,17 @@ export function getProfiles(page = 0, perPage: number = General.PROFILE_CHUNK_SI
 
 export function getMissingProfilesByIds(userIds: string[]): ActionFunc {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        const {profiles} = getState().entities.users;
+        const state = getState();
+        const {profiles} = state.entities.users;
+        const enableUserStatuses = getIsUserStatusesConfigEnabled(state);
         const missingIds: string[] = [];
         userIds.forEach((id) => {
             if (!profiles[id]) {
                 missingIds.push(id);
             }
         });
 
-        if (missingIds.length > 0) {
+        if (missingIds.length > 0 && enableUserStatuses) {
             getStatusesByIds(missingIds)(dispatch, getState);
             return getProfilesByIds(missingIds)(dispatch, getState);
```

Comment:
Shouldn't this be
```suggestion
        if (missingIds.length > 0) {
            if (enableUserStatuses) {
            	getStatusesByIds(missingIds)(dispatch, getState);
            }
            return getProfilesByIds(missingIds)(dispatch, getState);
```

Because right now, we won't get the missing profiles if the user status is disabled - which seems weird.

Reply:
Exactly

Reply:
awesome, excellent catch, thanks @JulienTant